lib/identifiers/isbn.rb

```diff
@@ -38,24 +40,20 @@ class ISBN
       /         # Prefix/suffix divider
       \d{1,7}   # ISBN title enumerator and check digit
       \b
-    }x
-    TEXT_AFTER_PREFIX_REGEXP = ':?\s*(\d.*)$'.freeze
+    }x.freeze
 
-    def self.extract(str , prefixes = [])
-      str = match_strings_with_prefix(str , prefixes) if prefixes.any?
+    def self.extract(str, prefixes = [])
+      return extract_with_prefix(str, prefixes) if prefixes.any?
 
       extract_isbn_as(str) + extract_thirteen_digit_isbns(str) + extract_ten_digit_isbns(str)
     end
 
-    def self.match_strings_with_prefix(str, prefixes)
-      prefix_regexp = prefixes.join('|')
+    def self.extract_with_prefix(str, prefixes)
+      prefix_regexp = generate_prefix_regexp(prefixes)
 
-      str
-        .to_s
-        .scan(/(#{prefix_regexp})#{TEXT_AFTER_PREFIX_REGEXP}/i)
-        .inject('') do |acum, (_prefix, match)|
-          acum + "#{match} \n "
-        end
+      [isbn_a_candidate_matcher, ISBN_13_REGEXP, ISBN_10_REGEXP].inject([]) do |matches, isbn_regexp|
```

Comment:
I tried joining them in the same regexp but it is way more involved.
This solution is cleaner and easy to understand IMHO